front/components/agent_builder/utils.ts

```diff
@@ -20,3 +30,38 @@ export function getUniqueTemplateTags(
     new Set(templates.flatMap((template) => template.tags))
   ).sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
 }
+
+export function validateDataSourceConfiguration(
+  config: unknown
+): Result<DataSourceConfiguration, Error> {
+  try {
+    const validated = dataSourceConfigurationSchema.parse(config);
+    return new Ok(validated);
+  } catch (error) {
+    return new Err(normalizeError(error));
+  }
+}
+
+// Create minimal objects with placeholder values that will be populated by the server
+export function createMinimalDataSourceView(sId: string): DataSourceViewType {
+  return {
+    sId,
+    id: 0 as ModelId,
+    category: "managed" as DataSourceViewCategory,
```

Comment:
nit: I'd use `satisfies` here instead of `as`. If we later change the options in `DataSourceViewCategory` we'd get a nice typescript error if "managed" is not present.